src/NServiceBus.Core/EndpointInstance.cs

```diff
@@ -1,21 +1,33 @@
 ﻿namespace NServiceBus
 {
-    using System;
-    using System.Text;
+    using System.Collections.Generic;
+    using System.Linq;
 
     /// <summary>
     /// Represents a name of an endpoint instance.
     /// </summary>
     public sealed class EndpointInstance
     {
+        readonly PropertyDictionary properties;
+        
+        /// <summary>
+        /// Returns the name of the endpoint.
+        /// </summary>
+        public Endpoint Endpoint { get; }
+
+        /// <summary>
+        /// A specific discriminator for scale-out purposes.
+        /// </summary>
+        public string ScaleOutDiscriminator { get; }
```

Comment:
I we are representing scale out concepts in EndpointInstance maybe we should pull here qualifer from LocalAddress and call it SatelliteDiscriminator (which might be more expressive)?